File: app/src/main/kotlin/tech/mmarca/openvitals/core/period/PeriodRangePreferenceKey.kt
package tech.mmarca.openvitals.core.period

enum class PeriodRangePreferenceKey(
    val storageKey: String,
    val defaultRange: TimeRange,
) {
    STEPS("detail_range_steps", TimeRange.WEEK),
    CALORIES("detail_range_calories", TimeRange.WEEK),
    ACTIVITIES("detail_range_activities", TimeRange.WEEK),
    SLEEP("detail_range_sleep", TimeRange.WEEK),
    HEART("detail_range_heart", TimeRange.WEEK),
    BODY("detail_range_body", TimeRange.MONTH),
    HYDRATION("detail_range_hydration", TimeRange.WEEK),
    NUTRITION("detail_range_nutrition", TimeRange.WEEK),
    MINDFULNESS("detail_range_mindfulness", TimeRange.WEEK),
    CYCLE("detail_range_cycle", TimeRange.MONTH),
}

File: lib/core/period/period_range_preference_key.dart
import 'time_range.dart';

/// The persisted last-selected [TimeRange] per detail/list screen.
enum PeriodRangePreferenceKey {
  steps('detail_range_steps', TimeRange.week),
  calories('detail_range_calories', TimeRange.week),
  activities('detail_range_activities', TimeRange.week),
  sleep('detail_range_sleep', TimeRange.week),
  heart('detail_range_heart', TimeRange.week),
  body('detail_range_body', TimeRange.month),
  hydration('detail_range_hydration', TimeRange.week),
  nutrition('detail_range_nutrition', TimeRange.week),
  mindfulness('detail_range_mindfulness', TimeRange.week),
  // A month, not a week: recovery tests are not a daily thing, and a week of them is
  // usually one point or none.
  heartRateRecovery('detail_range_hrr', TimeRange.month),
  cycle('detail_range_cycle', TimeRange.month);

  const PeriodRangePreferenceKey(this.storageKey, this.defaultRange);

  final String storageKey;
  final TimeRange defaultRange;
}
